Add enabled flag to the public file server settings.

kaspth opened the pull request:
As discussed in https://github.com/rails/rails/pull/19135#issuecomment-153385986.

Replaces `serve_static_files` to unify the static options under the `public_file_server` wing.

Deprecates `serve_static_files` accessors, but make them use the newer config internally.

kaspth committed:
Add enabled flag to the public file server.

As discussed in https://github.com/rails/rails/pull/19135#issuecomment-153385986.

Replaces `serve_static_files` to unify the static options under the `public_file_server` wing.

Deprecates `serve_static_files` accessors, but make them use the newer config internally.

rafaelfranca commented:
:shipit:

kaspth commented:
Sweet! Will ship it in the morning, don't want to accidentally leave the build broken overnight :grin:

Also planning of pulling over one of the changelog entries to Railties: https://github.com/rails/rails/pull/19135#discussion_r43815066

schneems commented:
Since we've got 3 different flags for this now. Can we make a different flag for assuming the public folder won't change after boot so we can effectively cache all files at start up and not have to hit disk on every request?

kaspth committed:
Move `static_cache_contorl` deprecation changelog entry to Railties.

The configuration for `config.static_cache_control`, and its replacement
`config.public_file_server.headers` are implemented in Railties.

People would configure this in environment files, which is Railties domain too.

kaspth commented:
@schneems yeah, that sounds cool!

Could add an `eager_load` option to `config.public_file_server`, which just mimics `config.eager_load` by default.
